modules/core/src/main/scala/org/tessellation/infrastructure/rewards/Rewards.scala

```diff
@@ -21,30 +24,31 @@ import eu.timepit.refined.auto._
 import eu.timepit.refined.numeric.Positive
 import eu.timepit.refined.refineV
 import io.estatico.newtype.ops.toCoercibleIdOps
+import org.typelevel.log4cats.slf4j.Slf4jLogger
 
 object Rewards {
 
   def make[F[_]: Async: SecurityProvider](
-    config: RewardsConfig,
+    rewardsPerEpoch: SortedMap[EpochProgress, Amount],
     softStaking: SimpleRewardsDistributor,
     dtm: SimpleRewardsDistributor,
     stardust: SimpleRewardsDistributor,
     regular: RewardsDistributor[F]
   ): Rewards[F] =
     new Rewards[F] {
 
-//      private val logger = Slf4jLogger.getLogger[F]
+      private val logger = Slf4jLogger.getLogger[F]
 
       def calculateRewards(
         epochProgress: EpochProgress,
         facilitators: NonEmptySet[Id]
       ): F[SortedSet[RewardTransaction]] = {
-        val amount = getAmountByEpoch(epochProgress)
+        val amount = getAmountByEpoch(epochProgress, rewardsPerEpoch)
 
         val programRewardsState = for {
-          dtmRewards <- dtm.distribute(epochProgress, facilitators)
-          softStakingRewards <- softStaking.distribute(epochProgress, facilitators)
           stardustCollective <- stardust.distribute(epochProgress, facilitators)
+          softStakingRewards <- softStaking.distribute(epochProgress, facilitators)
+          dtmRewards <- dtm.distribute(epochProgress, facilitators)
```

Comment:
Nice that you've caught it. I was prototyping, and at some point, I reversed the order to use composition instead of sequencing, and then I forgot to revert it back.